crux-bench/src/crux/bench/main.clj

```diff
@@ -11,15 +11,24 @@
   (bench/post-to-slack (format "*Starting Benchmark*, Crux Version: %s, Commit Hash: %s\n"
                                bench/crux-version bench/commit-hash))
 
-  (bench/with-nodes [node]
-    (let [devices-results (devices/run-devices-bench node)
-          weather-results (weather/run-weather-bench node)
-          watdiv-results (watdiv-crux/run-watdiv-bench node {:test-count 100})
-          result-messages [(bench/results->slack-message devices-results :ts-devices)
-                           (bench/results->slack-message weather-results :ts-weather)
-                           (bench/results->slack-message
-                            [(first watdiv-results) (watdiv-crux/->query-result (rest watdiv-results))]
-                            :watdiv-crux)]]
-      (bench/send-email-via-ses (string/join "\n" result-messages))))
+  (->> (bench/with-nodes [node]
+         (let [devices-results (doto (devices/run-devices-bench node)
+                                 (bench/post-slack-results :ts-devices))
+
+               weather-results (doto (weather/run-weather-bench node)
+                                 (bench/post-slack-results :ts-weather))
+
+               watdiv-results (doto (watdiv-crux/run-watdiv-bench node {:test-count 100})
+                                (bench/post-slack-results :watdiv-crux))]
+           [devices-results
+            weather-results
+            watdiv-results]))
+
+       (reduce (fn [acc [node-type result]]
```

Comment:
you don't need the power of a `reduce` here, map + string/join is sufficient. I'd try to steer clear of `reduce` unless later values need the accumulator - as a reader, I need to reason about how the elements might interact with each other; with a `map`, I _know_ they can't.